### PR TITLE
Add debug logs for terminated, scheduled workflow executor

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
This change adds logging and optional reconnect attempts for the scheduled workflow executor to address https://github.com/lyft/flyte/issues/198

This is step 1 in diagnosing the problem (it's difficult to reproduce locally). Step 2 is adding behavior to handle the observed failure causes :)

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/198

## Follow-up issue
https://github.com/lyft/flyte/issues/88

### CODE REVIEW

**Comment by katrogan** [comment]
cc @rstanevich

File: pkg/rpc/adminservice/base.go [hunk 1]
```diff
@@ -135,8 +138,17 @@ func NewAdminServer(kubeConfig, master string) *AdminService {
...
 	logger.Info(context.Background(), "Successfully initialized a new scheduled workflow executor")
 	go func() {
+		logger.Info(context.Background(), "Starting the scheduled workflow executor")
 		scheduledWorkflowExecutor.Run()
-		logger.Info(context.Background(), "Successfully started running the scheduled workflow executor")
+
+		maxReconnectAttempts := configuration.ApplicationConfiguration().GetSchedulerConfig().
+			WorkflowExecutorConfig.ReconnectAttempts
+		for reconnectAttempt := 0; reconnectAttempt < maxReconnectAttempts; reconnectAttempt++ {
+			time.Sleep(workflowExecutorReconnectDelay)
```
Comment (EngHabu) [inline pkg/rpc/adminservice/base.go:147]: Why is the delay? do we need to add Jitter?

File: pkg/rpc/adminservice/base.go [hunk 2]
```diff
@@ -50,6 +51,8 @@ func (m *AdminService) interceptPanic(ctx context.Context, request proto.Message
 
 const defaultRetries = 3
 
+var workflowExecutorReconnectDelay = 30 * time.Second
```
Comment (EngHabu) [inline pkg/rpc/adminservice/base.go:54]: If we are making the number of reconnect attempts configurable, should we make this too?

Comment (katrogan) [reply pkg/rpc/adminservice/base.go:54]: sure

File: pkg/rpc/adminservice/base.go (hunk 1 above)
Comment (katrogan) [reply pkg/rpc/adminservice/base.go:147]: yes, but it's just an (unproven) idea. i have no idea what's causing the client connection to terminate so i figured it might be useful. if you disagree i can remove altogether

File: pkg/rpc/adminservice/base.go [hunk 3]
```diff
@@ -135,8 +138,17 @@ func NewAdminServer(kubeConfig, master string) *AdminService {
 	scheduledWorkflowExecutor := workflowScheduler.GetWorkflowExecutor(executionManager, launchPlanManager)
 	logger.Info(context.Background(), "Successfully initialized a new scheduled workflow executor")
 	go func() {
+		logger.Info(context.Background(), "Starting the scheduled workflow executor")
 		scheduledWorkflowExecutor.Run()
```
Comment (kumare3) [inline pkg/rpc/adminservice/base.go:142]: do you want to remove this one?

File: pkg/rpc/adminservice/base.go (hunk 1 above)
Comment (kumare3) [reply pkg/rpc/adminservice/base.go:147]: does it connect to the pubsub, on AWS SQS?

File: pkg/rpc/adminservice/base.go (hunk 3 above)
Comment (katrogan) [reply pkg/rpc/adminservice/base.go:142]: I wanted to support the default where no reconnect attempts are specified, but I can if you think this is cleaner

File: pkg/rpc/adminservice/base.go (hunk 1 above)
Comment (katrogan) [reply pkg/rpc/adminservice/base.go:147]: yeah

**Comment by katrogan** [comment]
PTAL @EngHabu

**Comment by rstanevich** [comment]
@katrogan
Thanks you a lot for this fix!
So, more than one month we have no any interrupting with scheduled workflows :) 

There are a few errors from Gizmo client like
```
{"json":{"src":"workflow_executor.go:246"},"level":"error","msg":"Gizmo subscriber closed channel with err: [RequestError: send request failed\ncaused by: Post https://sqs.us-east-1.amazonaws.com/: read tcp read: connection reset by peer]","ts":"2020-05-26T15:36:50Z"}
```
But subscriber restarts gracefully as I see.
